webroot/js/directives/sentence-and-translations.dir.js

```diff
@@ -283,6 +336,102 @@
                 vm.sentence.isOwnedByCurrentUser = vm.sentence.user && vm.sentence.user.username;
             });
         }
+
+        function list() {
+            initLists(vm.sentence.sentences_lists);
+
+            if (vm.visibility['list_form']) {
+                closeList();
+            } else {
+                show('list_form');
+                focusInput('#list-form-' + vm.sentence.id);
+                resetLists();
+            }
+        }
+
+        function closeList() {
+            vm.hide('list_form');
+            vm.listSearch = '';
+        }
+
+        function toggleList(list) {
+            vm.inProgress = true;
+            var action = list.hasSentence ? 'add_sentence_to_list' : 'remove_sentence_from_list';
+
+            $http.get(rootUrl + '/sentences_lists/' + action + '/' + vm.sentence.id + '/' + list.id).then(function(result) {
+                vm.inProgress = false;
+                $cookies.put('most_recent_list', list.id);
+            });
+        }
+
+        function addToNewList() {
+            if (!vm.listSearch) {
+                return;
+            }
+            
+            vm.inProgress = true;
+            var data = { 
+                name: vm.listSearch,
+                sentenceId: vm.sentence.id
+            };
+            $http.post(rootUrl + '/sentences_lists/add_sentence_to_new_list', data).then(function(result) {
+                vm.inProgress = false;
+                vm.listSearch = '';
+                $rootScope.$broadcast('newListCreated', result.data.result, vm.sentence.id);
```

Comment:
If it fails, the action `add_sentence_to_new_list` will return the value `{result: 'error'}` in `result.data` (as opposed to the list data). It looks like you are not checking for that case.